tools/src/bin/scip-indexer.rs

```diff
@@ -774,7 +855,12 @@ fn analyze_using_scip(
                 // so only process the first element.
```

Comment:
This does not work as expected with Java's interfaces which enables some kind of multiple inheritance.
Take for instance [org.webrtc.AndroidVideoDecoder](https://dev.searchfox.org/mozilla-central/search?q=symbol:S_java_java_org%2Fwebrtc%2FAndroidVideoDecoder%23). It implements both `VideoDecoder` and `VideoSink` but only the former gets picked up here.

There is worse though, for [org.mozilla.thirdparty.com.google.android.exoplayer2.ExoPlayerImpl](https://dev.searchfox.org/mozilla-central/search?q=symbol:S_java_java_org%2Fmozilla%2Fthirdparty%2Fcom%2Fgoogle%2Fandroid%2Fexoplayer2%2FExoPlayerImpl%23) which has this class diagram:
```mermaid
classDiagram
    class Player
    <<interface>> Player

    class ExoPlayer
    <<interface>> ExoPlayer

    class BasePlayer
    <<abstract>> BasePlayer

    class ExoPlayerImpl
    <<final>> ExoPlayerImpl

    BasePlayer <|-- ExoPlayerImpl : extends
    ExoPlayer <|-- ExoPlayerImpl : implements

    Player <|-- ExoPlayer : extends

    Player <|-- BasePlayer : implements
```
The only superclass of `ExoPlayerImpl` picked up by Searchfox here is `Player`. Iow the order in which the relationships appear in scip_sym_info is not guaranteed to be from closest to farthest.

The cleanest fix would be to have the SCIP data distinguish between direct parents and other ancestors − or provide an option to only list direct parents.

Reply:
> The cleanest fix would be to have the SCIP data distinguish between direct parents and other ancestors − or provide an option to only list direct parents.

Agreed.  I also think that while it would be great if SCIP-producing upstreams expressed this nuance, it may not realistically be something we can depend on or require of upstreams as I think for the general VS code language server use case there is no need to care, even though searchfox might care.

The crossref mechanism can potentially undo the transitive inclusion process here.  Perhaps we could use the "props" array to include something like "possibly-transitive" in cases when we don't know if the edge is direct or transitive.  crossref could then eventually process these edges, normalizing them by retaining the edge on cases that seem direct (and removing the "possibly-transitive" prop) and removing the edges that seem like they're only included because they're transitive.  I don't think we need to do that in this PR, but it would be nice as a follow-up[1].

In particular, I don't believe the "search" endpoint would be bothered by the transitive closure here, although it does make it more likely that the search results will be overwhelming.  It's only experimental things like [inheritance-diagram diagrams](https://asuth.searchfox.org/mozilla-central/query/default?q=inheritance-diagram%3Aworkercontrolrunnable+graph-layout%3Aneato) and `crossref_expand` (which currently is not usable both because the "query" endpoint UI needs improvement, but also because I think it has bugs) which would care.

Reply:
To clarify, I think it's fine to remove this constraint that we only process the first element in this PR.  Also fine to do it in another PR, but I think the upsides of having the full set of direct parents probably outweighs the downsides of having to deal with the full transitive set.